### ISSUE TITLE
'flutter run -d emulator' Runs the Old Version of App

### ISSUE THREAD

**Issue by KeroppiMomo** [flutter/flutter#36060]
## Steps to Reproduce
1. Run `flutter run -d emulator`.
2. It will run the old version of the app. Only after pressing `R` (hot restarting) will it run the current version.

## Logs

`flutter run --verbose`: https://pastebin.com/rvkuUcw7

```
$ flutter analyze
Analyzing [app_name]...                                        
No issues found! (ran in 4.0s)
```

```
$ flutter doctor -v
[✓] Flutter (Channel master, v1.8.1-pre.23, on Mac OS X 10.14.5 18F132, locale zh-Hant-HK)
    • Flutter version 1.8.1-pre.23 at /Users/[myself]/flutter
    • Framework revision 3bf91b5436 (7 days ago), 2019-07-05 12:59:32 -0700
    • Engine revision 3c51a7bfff
    • Dart version 2.5.0 (build 2.5.0-dev.0.0 b5aeaa6796)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/[myself]/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • CocoaPods version 1.7.3

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 35.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.36.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.2.0

[✓] Connected device (5 available)
    • Android SDK built for x86 • emulator-5554                        • android-x86    • Android Q (API 28) (emulator)
    • iPhone 5s                 • 4B7CC200-BCB9-4289-BA97-0C5BB4FCB4B2 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-12-2 (simulator)
    • iPhone Xs Max             • 3C563C80-D1C8-4A2C-AD39-23534419043F • ios            • com.apple.CoreSimulator.SimRuntime.iOS-12-2 (simulator)
    • macOS                     • macOS                                • darwin-x64     • Mac OS X 10.14.5 18F132
    • Chrome                    • chrome                               • web-javascript • Google Chrome 75.0.3770.100 


• No issues found!
```

**Comment by jonahwilliams** [comment]
> It will run the old version of the app.

When you say the old version of the app, this is re-running after already having built and made some changes?

**Comment by KeroppiMomo** [comment]
After I uninstall and `flutter run` the app (e.g. showing text "A" on screen), it shows the correct result ("A"). However, if I change the code to show "B" and `flutter run` the app again, it still shows "A" instead of "B". It only shows "B" after hot restarting. I think the built APK file contains the old version of the app.

**Comment by eseidelGoogle** [comment]
Sounds like the app is failing to install on the emulator.

Are you still seeing this with more recent versions of Flutter (we continue to make improvements to the tooling)?

If you are, could you please add `flutter run -v` to help us diagnose?

Thanks!

**Comment by millerovv** [comment]
Had the same problem recently. ```flutter clean``` helped, but after that I also had this problem:
https://stackoverflow.com/questions/57980900/flutter-could-not-find-the-built-application-bundle-at-build-ios-iphonesimulato
after fixing which by following the top answer everything worked just fine

**Comment by jonahwilliams** [comment]
essentially a duplicate of https://github.com/flutter/flutter/issues/16604 . The stackoverflow issue is separate
